### PR TITLE
fix unbound variable issue when get version

### PR DESCRIPTION
**Pull request by hchenxa**
there have variable unbound issues when use get scripts like below:

./get: line 197: 2: unbound variable

### CODE REVIEW

**Comment by jascott1** [comment]
Doesn't this allow `--version` to be specified without passing a version and then fall back to latest? It seems the improved behavior would throw version required error when flag was present without value. What do you think?

**Comment by hchenxa** [comment]
@jascott1 from the function help(), it seems the version should be provided when use -v|--version like below:
```
help () {
  echo "Accepted cli arguments are:"
  echo -e "\t[--help|-h ] ->> prints this help"
  echo -e "\t[--version|-v <desired_version>] . When not defined it defaults to latest"
  echo -e "\te.g. --version v2.4.0  or -v latest"
}
```
and if the --version|-v did not specified, the latest will be used.

File: scripts/get [hunk 1]
```diff
@@ -191,9 +191,14 @@ export INPUT_ARGUMENTS="${@}"
...
   case $1 in
-    '--version'|-v) 
-       export DESIRED_VERSION="${2}" 
+    '--version'|-v)
        shift
+       if [[ $# -ne 0 ]]; then
+           export DESIRED_VERSION="${1}"
+       else
+           echo -e "Please provide the desired version. e.g. --verison v2.4.0 or -v latest"
+           exit 0
```
Comment (jascott1) [inline scripts/get]: This should exit with non-zero code

**Review by jascott1 — CHANGES_REQUESTED** [review]

File: scripts/get (hunk 1 above)
Comment (hchenxa) [reply scripts/get]: @jascott1 , seems it non-zero code returned, fail_trap() function will be called and output will like below:
```
Please provide the desired version. e.g. --verison v2.4.0 or -v latest
Failed to install helm with the arguments provided: -v
Accepted cli arguments are:
	[--help|-h ] ->> prints this help
	[--version|-v <desired_version>] . When not defined it defaults to latest
	e.g. --version v2.4.0  or -v latest
	For support, go to https://github.com/kubernetes/helm.
```

Please help to confirm if this was the correct behavior?

Comment (jascott1) [reply scripts/get]: Sorry I was just thinking exit code and blanked on the trap. Lets keep it 0 and return the version required message when the flag is specified without value. Thanks @hchenxa !

Comment (hchenxa) [reply scripts/get]: @jascott1 , any comments on message output when the desired version not provided:
```
Please provide the desired version. e.g. --version v2.4.0 or -v latest
```

**Review by gyliu513 — COMMENTED** [review]
/lgtm

**Comment by bacongobbler** [comment]
removing the lgtm label since this hasn't been officially approved by a [core maintainer](https://github.com/kubernetes/helm/blob/master/OWNERS) as per [our CONTRIBUTING docs](https://github.com/kubernetes/helm/blob/master/CONTRIBUTING.md#pull-requests).

File: scripts/get (hunk 1 above)
Comment (jascott1) [reply scripts/get]: Looks great. Thanks!

**Review by jascott1 — APPROVED** [review]

File: scripts/get [hunk 2]
```diff
@@ -191,9 +191,14 @@ export INPUT_ARGUMENTS="${@}"
...
 while [[ $# -gt 0 ]]; do
   case $1 in
-    '--version'|-v) 
-       export DESIRED_VERSION="${2}" 
+    '--version'|-v)
        shift
+       if [[ $# -ne 0 ]]; then
+           export DESIRED_VERSION="${1}"
+       else
+           echo -e "Please provide the desired version. e.g. --verison v2.4.0 or -v latest"
```
Comment (bacongobbler) [inline scripts/get]: typo: --verison should be --version.

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: scripts/get (hunk 2 above)
Comment (jascott1) [reply scripts/get]: Thanks for the catch!

**Comment by hchenxa** [comment]
@bacongobbler , thanks for review and the typo issue has gone.

**Comment by hchenxa** [comment]
any other feedback @bacongobbler @jascott1? Can we have this got approved?

**Review by bacongobbler — APPROVED** [review]
Thanks for the bump!

**Pull request merged**